crypto/defaults.c

```diff
@@ -69,8 +69,8 @@ static char *get_windows_regdirs(char *dst, LPCTSTR valuename)
     HKEY hkey;
     LSTATUS ret;
     DWORD index = 0;
-    LPCTCH tempstr = NULL;
-   
+    LPBYTE tempstr = NULL;
+
```

Comment:
not sure I understand the need for this, as its throwing a warning during build:
```
warning C4133: 'function': incompatible types - from 'LPBYTE' to 'LPCWCH'
```
LPTCH is the type that matches the call to the argument in WideCharToMultiByte

Reply:
Indeed that is probably not the correct type, although without the change for me it straight up fails due to `LPCTCH` not being a known type
On a 2nd thought, it seems like this could be due to my command line lacking `-DUNICODE`, but AFAIU it should be automatically included when building for `mingw64`

Reply:
Then the right solution would be to include a patch to include that define for that platform

Reply:
Oh no that's my bad, `-DUNICODE` is there, and if I understand mingw64'h headers correctly, `LPCTCH` will only be available when `UNICODE` isn't defined.

Wouldn't `LPSTR` be a better choice for `tempstr` as it's the type expected by `WideCharToMultiByte`, `LPCTCH` being a type alias to the former.

Reply:
According to this:
https://learn.microsoft.com/en-us/windows/win32/intl/windows-data-types-for-strings

LPSTR is typed to TCHAR same as LPCTCH, so I don't see that changing anything (unless mingw has some bug in its header includes).  LPCWCH might make more sense.

Reply:
Yes, AFAICS mingw-w64 only defines `LPCTCH` when `UNICODE` is defined (https://github.com/mingw-w64/mingw-w64/blob/master/mingw-w64-headers/include/winnt.h#L395 is the only definition available), which looks like a bug

For the time being I went with `LPCTSTR` which should be equivalent and works with both `UNICODE` defined or not

Reply:
I've changed `tempstr` and `valuename` to `LPWSTR` so I could use the `W` version of `RegQueryValueExW`. Previously, if the code was built in ANSI mode then `RegQueryValueExA` would be called, return an ANSI encoded string, and the call to `WideCharToMultiByte` would be incorrect.

Always calling `RegQueryValueExW` allows us to avoid handling that complexity and avoid the extra conversions performed internally by `RegQueryValueExA`.